### ISSUE TITLE
Xcode 12.5: Umbrella header for module 'Amplitude' does not include header '<...>'

### ISSUE THREAD

**Issue by abesuunto** [amplitude/Amplitude-iOS#342]
We are using the library via SPM in Xcode.

Xcode version: Xcode 12.5b3

- Build settings:

   - Swift compiler - warning policies:
  treat warnings as error: YES

## Expected Behavior

The library should build correctly.

## Current Behavior

Library fails to build because of the following warnings:

```
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'ISPPinnedNSURLSessionDelegate.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPURLConnection.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPDeviceInfo.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPDatabaseHelper.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPConstants.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'ISPPinnedNSURLConnectionDelegate.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPBackgroundNotifier.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AmplitudePrivate.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'Amplitude+SSLPinning.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPConfigManager.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPUtils.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'AMPURLSession.h'
/<module-includes>:1:9: In file included from <module-includes>:1:
/Users/<..>/SourcePackages/checkouts/Amplitude-iOS/Sources/Amplitude/Amplitude.h:689:1: Umbrella header for module 'Amplitude' does not include header 'ISPCertificatePinning.h'
```

## Possible Solution

Maybe this: https://forums.swift.org/t/root-cause-of-umbrella-header-for-module-does-not-include/24063

Fix build warnings that mentioned above for the Amplitude iOS SDK SPM 

## Steps to Reproduce
 

1. Create a project that pulls in the library from SPM.
2. Change Xcode builds settings > Swift compiler - warning policies > treat warnings as error: YES
3. Try to build with Xcode 12.5b3

## Environment
- SDK Version: 8.2.1
- Xcode: 12.5b3
- OS Version: macOS Big Sur 11.2.3

**Comment by yuhao900914** [comment]
Hi @abesuunto 
This issue has been fixed in [#335 ](https://github.com/amplitude/Amplitude-iOS/pull/335). It may be because of the Swift Package Cache issue. Please `Reset package caches` or remove and re-add. Then try to build. 
Thanks.

**Comment by abesuunto** [comment]
This is not the same issue as in #335. Please, repeat the steps I have mentioned. If I change `treat warnings as error:NO` when building it works.

**Comment by yuhao900914** [comment]
Got it now. 
cc: @jooohhn

**Comment by abesuunto** [comment]
Is there any plans to fix it soon? Xcode 12.5 release is around the corner. I think it is quite common to use `treat warnings as error: YES` setting in iOS projects. It ensures better quality code.

**Comment by yuhao900914** [comment]
@abesuunto  We are working on that now. Will give you an update soon.

**Comment by abesuunto** [comment]
Great! Thank you very much!

**Comment by cmachadomarcos** [comment]
Hello! 
any update? we are blocking the update to Xcode 12.5 because of that :)

**Comment by gmorning** [comment]
As Xcode 12.5 is mandatory to run apps on released iOS 14.5, this issue is becoming more and more severe each day. If you can increase it's priority, I'm kindly asking to do that

**Comment by yuhao900914** [comment]
Thanks for all of your patient. This issue has been fixed on the updated version v8.3.0. I Will close this issue due to it has been fixed.

**Comment by cmachadomarcos** [comment]
Thank you so much!
